src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSource.cs

```diff
@@ -56,7 +56,7 @@ public abstract partial class DiagnosticSource
         /// Null by default. Consumers should expect to receive null which may indicate that producer
         /// called pure IsEnabled(string) or producer passed all necessary context in arg1</param>
         /// <seealso cref="IsEnabled(string)"/>
-        public virtual bool IsEnabled(string name, object arg1, object arg2 = null)
+        public virtual bool IsEnabled(string name, object? arg1, object? arg2 = null)
```

Comment:
Can/should the name be nullable here?

Reply:
IsEnabled(name); is throwing on null

Reply:
> IsEnabled(name); is throwing on null

Can you share the example you're using?

Reply:
```csharp
DiagnosticSource httpLogger = new DiagnosticListener(null);
if (httpLogger.IsEnabled(null))
    httpLogger.Write("RequestStart", new { Url = "http://clr" });
```
It is throwing NRE on row httpLogger.IsEnabled(null), not on constructer with null argument